### DISCUSSION TITLE
How does mini batching help with dealing with variable sized input graphs

### DISCUSSION THREAD

**Discussion by VHK459** [pyg-team/pytorch_geometric#8133] [Q&A] [answered]
I know that if input graph size is variable then we use minibatching to make a single giant block diagonal matrix consisting of all graphs.
But as show in this  [Graph Classification.ipynb]( https://colab.research.google.com/drive/1I8a0DfQ3fI7Njc62__mVXUlcAleUclnb#scrollTo=xs5D713Ia_Sv) example by pytorch-geometric the total size of each batch is not same,(as no. of graphs is same ) . So how is GCNConv handling this variability in input. As GCN is $` \hat{X} = \tilde{D}^{-\frac{1}{2}} \tilde{A} \tilde{D}^{-\frac{1}{2}} X W `$ where **W** is the projection matrix. As here size of input is variable then how does GCN use **W** matrix.

**Comment by rusty1s** [comment, marked as answer]
The weight matrix is independent of the size of nodes. It is just a matrix mapping a node feature to a transformed version of it, which is shared across all nodes. Does that answer your question?

**Reply by VHK459** [reply]
Ok, so we first calculate  $` \bar{X} = \tilde{D}^{-\frac{1}{2}} \tilde{A} \tilde{D}^{-\frac{1}{2}} X  `$ then multiply $`\bar{X} W `$ to get $` \hat{X}`$ for all nodes and then do mean pooling to get graph embeddings.  Correct ?

**Reply by rusty1s** [reply]
That's correct.
